### PR TITLE
restructuring to hold the enum

### PR DESCRIPTION
**Pull request by gilescope**

### CODE REVIEW

File: ss58-registry-derive/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+[package]
+name = "ss58-registry-derive"
```
Comment (bkchr) [inline ss58-registry-derive/Cargo.toml:2]: This is not a derive macro.

And do we really need an extra proc macro for this? Why isn't that a build.rs?

File: ss58-registry-derive/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+version = "0.1.0"
+edition = "2018"
+description = "Registry of wellknown SS58 id types"
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
```
Comment (bkchr) [inline ss58-registry-derive/Cargo.toml:7]: Can you please not commit such stuff that is auto generated by cargo.

File: ss58-registry-derive/Cargo.toml (hunk 1 above)
Comment (gilescope) [reply ss58-registry-derive/Cargo.toml:2]: I don't think there's much difference between a proc macro and a build.rs. They both require a rustc invocation and then are executed. I think it's more honest to do this as a proc macro than to try and shoehorn it into a build.rs. I agree it's not a custom derive so I should call it something else.

Comment (bkchr) [reply ss58-registry-derive/Cargo.toml:2]: This also means one less crate that pollutes crates.io for no reason. A lot of projects are doing this, e.g. prost. I would also favor this approach.

**Pull request merged**